### PR TITLE
Js translation

### PR DESCRIPTION
**Pull request by lxzhangchao**
Translate the JS、JSX file.

### CODE REVIEW

**Comment by lxzhangchao** [comment]
I saw in the 'superset/assets/javascripts/ explore/actions/exploreActions js' in' $= window. $= the require (' jquery ') 'can be used, why can't use in 'superset/assets/javascripts/explore/stores/getLanguage.js', there was' ReferenceError: the window is not defined' mistakes, where I need to configure it.
When I set '$= window.$= require' (' jquery ') 'to' $= require '(' jquery'), it prompts that '$. Ajax is not a function'！
can you help me!

**Comment by xrmx** [comment]
Why are you opening a new PR?

**Comment by lxzhangchao** [comment]
I want to create a new PR to remove excess commit, so I just open a branch and submit a new PR, think excess commit will disappear, the results still exist excess commit.I'm Sorry!

**Comment by xrmx** [comment]
That's not how git works :) google for git rebase.
Once you have a handful of commits you can do the same after updating your master branch with latest changes and deal with conflicts.

**Comment by lxzhangchao** [comment]
I am very difficult to get the ajax request in getLanguage and can‘t get the current language. Do you have any Suggestions?

File: superset/assets/javascripts/locales.jsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
+import Jed from 'jed';
+import React from 'react';
+import { sprintf } from 'sprintf-js';
+import { getTranslations } from './translations';
+import { getLanguage } from './explore/stores/getLanguage';
+
+let LOCALE_DEBUG = false;
+
+if (sessionStorage && sessionStorage.getItem('localeDebug') === '1') {
```
Comment (mistercrunch) [inline superset/assets/javascripts/locales.jsx]: What is `sessionStorage` and how does it make it to the global scope?

File: superset/assets/javascripts/locales.jsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    domain: 'superset',
+    locale_data: {
+      superset: translations,
+    },
+  });
+}
+
+setLocale(getLanguage());
+
+function formatForReact(formatString, args) {
```
Comment (mistercrunch) [inline superset/assets/javascripts/locales.jsx]: Why is `formatForReact` necessary?

File: superset/assets/webpack.config.js [hunk 3]
```diff
@@ -108,6 +108,14 @@ const config = {
         include: APP_DIR + '/node_modules/mapbox-gl/js/render/painter/use_program.js',
         loader: 'transform/cacheable?brfs',
       },
+      {
+        test: /\.po$/,
+        loader: 'po-catalog-loader',
```
Comment (mistercrunch) [inline superset/assets/webpack.config.js]: Is babel or `po-catalog-loader` collecting the strings? I thought the babel command was doing it. Do we need both?

**Review by mistercrunch — COMMENTED** [review]
a few more comments

**Comment by mistercrunch** [comment]
Would be nice if the PR didn't include all of the string translations... Makes it hard to read only the essence of it.

What exactly is the blocker with `getLanguage`?

File: superset/assets/javascripts/locales.jsx (hunk 2 above)
Comment (lxzhangchao) [inline superset/assets/javascripts/locales.jsx]: This method is used to parse the incoming parameters
such as 'title={t('Preview for %s', query.tableName)}'

File: superset/assets/webpack.config.js (hunk 3 above)
Comment (lxzhangchao) [inline superset/assets/webpack.config.js]: This is the PO file output that is parsed through Jed method. Jed requires po-catalog-loader, and I'm not sure if Babel can

**Comment by lxzhangchao** [comment]
You can see the code for only translating a page in 'https://github.com/apache/incubator-superset/compare/master...lxzhangchao:js_trans?expand=1‘’

**Comment by lxzhangchao** [comment]
Is there any way to get Superset current language in the JS file, to be able to compile it！

**Comment by lxzhangchao** [comment]
when execute to 'npm run sync-backend' command, it can't  ajax request or window,  the following error occurs:
qq 20170802140837
so the 'superset/assets/javascripts/explore/stores/controls.jsx' is not internationalization.

When you load the PO file with require, the following error occurs:

so I have executed some commands to convert the PO file into a JSON file.
Hope you can  receive this code and implement the Superset js internationalization.

**Comment by lxzhangchao** [comment]
Hope you can accept this code！ @xrmx

File: superset/assets/javascripts/locales.jsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+export function gettextComponentTemplate(template, components) {
+  const tmpl = parseComponentTemplate(i18n.gettext(template));
+  return renderComponentTemplate(tmpl, components);
+}
+
+export const t = gettext;
+export const tn = ngettext;
+export const tct = gettextComponentTemplate;
```
Comment (ciwei100000) [inline superset/assets/javascripts/locales.jsx]: Thanks for your contribution. May I ask in what condition tct will be used?

**Comment by mistercrunch** [comment]
Superseeded by https://github.com/apache/incubator-superset/pull/3390 that we're about to merge. Thanks for getting the ball rollin'!

**Pull request closed without merging**